File: hashing/01.Two Sum/twoSum.ts
function twoSumTs(nums: number[], target: number): number[] {
    let hash: { [key: number]: number } = {};
    for (let i = 0; i < nums.length; i++) {
        let diff = target - nums[i];
        if (diff in hash) {
            return [hash[diff], i];
        } else {
            hash[nums[i]] = i;
        }
    }
    return [-1, -1]
}

const res2 = twoSumTs([2, 7, 11, 12], 14)
console.log(res2);

File: hashing/01.Two Sum/twoSum.go
package main

import "fmt"

func main() {
	nums := []int{2, 7, 11, 12, 14}
	output := twoSum(nums, 14)
	fmt.Println(output)
}

func twoSum(nums []int, target int) [2]int {
	hashMap := make(map[int]int)
	var diff int
	for i, v := range nums {
		diff = target - v
		if v, found := hashMap[diff]; found {
			return [2]int{v, i}
		} else {
			hashMap[v] = i
		}
	}

	return [2]int{-1, -1}
}
